### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by aljones15**
Updates all of the tests to use a Mock Ed25519 Keypair for the invocationSigner, an X25519 key for the keyAgreementKey, and adds a method to assert on jwe headers.

### CODE REVIEW

File: tests/MockKaK.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+'use strict';
+
+const base58 = require('../base58');
+const nacl = require('tweetnacl');
+
+// ensures tests use the same KaK for each test.
+const _secretKey = new TextEncoder('utf-8').encode(
+  'testKaK0123456789testKaK01234567');
+
+export class MockKaK {
```
Comment (dlongley) [inline tests/MockKaK.js]:
```suggestion
export class MockKak {
```

File: tests/mock.js [hunk 2]
```diff
@@ -4,10 +4,9 @@
 import {DataHubClient} from '..';
 import {MockStorage} from './MockStorage.js';
 import {MockServer} from './MockServer.js';
-import {MockKek} from './MockKek.js';
 import {MockHmac} from './MockHmac.js';
-
-// FIXME use mock test data
+import {MockKaK} from './MockKaK.js';
```
Comment (dlongley) [inline tests/mock.js]:
```suggestion
import {MockKak} from './MockKak.js';
```

File: tests/mock.js [hunk 3]
```diff
@@ -23,25 +22,34 @@ class TestMock {
...
       // create mock keys
       this.keys = {};
-
-      // create KEK and HMAC keys for creating data hubs
-      this.keys.kek = await MockKek.create();
+      // this creates the same invocationSigner for each test.
+      this.invocationSigner = new MockInvoker();
+      // create KAK and HMAC keys for creating data hubs
+      // this creates the same keyAgreementKey for each test.
+      this.keys.keyAgreementKey = new MockKaK();
```
Comment (dlongley) [inline tests/mock.js]:
```suggestion
      this.keys.keyAgreementKey = new MockKak();
```

File: tests/MockKaK.js (hunk 1 above)
Comment (dlongley) [reply tests/MockKaK.js]: Filename needs to change too.

File: tests/mock.js [hunk 4]
```diff
@@ -23,25 +22,34 @@ class TestMock {
...
+      // the creates the same hmac for each test.
       this.keys.hmac = await MockHmac.create();
+      this.keyResolver = ({id}) => {
+        if(this.keys.keyAgreementKey.id === id) {
+          return this.keys.keyAgreementKey;
+        }
+        return this.keys.hmac;
```
Comment (dlongley) [inline tests/mock.js]: Why is this here? It seems like `throw new Error(`Key "${id}" not found.`);` would be better. My memory is that `keyResolver` functions are only for kaks, but I could be mistaken.

Comment (aljones15) [reply tests/mock.js]: I actually just left that as is, but yeah as far as I can tell it's only for Kaks.

Comment (aljones15) [reply tests/mock.js]: Addressed in https://github.com/digitalbazaar/secure-data-hub-client/pull/10/commits/be488030528e73efdf8f319b8d61e31036c59a7d

and I believe that is it.
